account-registration/src/main/java/com/amazon/aws/spinnaker/plugin/registration/AccountsStatus.java

```diff
@@ -275,17 +280,19 @@ private Response callApiGateway(String url) {
                 if (e instanceof HttpClientErrorException) {
                     HttpClientErrorException ex = (HttpClientErrorException) e;
                     if (HttpStatus.FORBIDDEN == ex.getStatusCode()) {
-                        log.error(e.getMessage());
-                        log.info("Received 403 from API Gateway.");
+                        log.info("Received 403 from API Gateway: {}", e.getMessage());
                         makeHeaderGenerator(url);
                         if (this.headerGenerator == null) {
                             log.error("Failed to generate resources required for AWS Signature V4 to authenticate with API Gateway.");
+                            break;
                         }
+                        retry += 1;
+                        continue;
```

Comment:
you can probably do it more cleanly with a `while(true)` as the condition and then doing a `break` out of the loop here. instead of  doing the count.